p2p/persist.go

```diff
@@ -46,7 +46,7 @@ func writePeers(h host.Host, dir string) error {
 	}
 
 	checksum := crc64.New(crc64.MakeTable(crc64.ISO))
-	tmp, err := os.CreateTemp(dir, "connected-***")
+	tmp, err := os.CreateTemp(dir, "connected.tmp")
```

Comment:
how it will work if connected.tmp exists? will it overwrite it?

Reply:
According to the documentation if no `*` are specified in the name, a random string is appended to its name. The problem was that `os.CreateTemp` only replaces the last `*` with a random string and `connected-**123972` is not a valid filename on windows.

So now the filename is `connected.tmp123798` where the number at the end is a different randomly generated one every time `os.CreateTemp` is called.

It would probably have been better if I would have used the pattern `connected-*.tmp` or similar to ensure it works on any platform. Do you want me to change it?